modules/flowable-engine-common-api/src/main/java/org/flowable/common/engine/api/tenant/ChangeTenantIdResult.java

```diff
@@ -0,0 +1,99 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ package org.flowable.common.engine.api.tenant;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Container class to return the result of a change in a tenant id operation (produced by a {@link ChangeTenantIdBuilder})
+ */
+public class ChangeTenantIdResult {
```

Comment:
This class does not follow the patterns for most of our APIs in Flowable. 

1. Let's make this an interface that would be able to provide a method that would return the number of changed instances for a particular key, all types of instances that were changed. Exposing the map might be OKish as well.
1. We shouldn't use an enum for the key as it makes it difficult to add new elements if we need them.
1. The public API doesn't need to have a builder exposed. The implementation class can easily accept a `Map<String, Long>` for the beginning.
1. equals and hashCode are not really needed. The important part is that the counts are correct. Is there a reason for these 2 methods, apart the use of them in the tests for the `isEqualTo` check

Reply:
I have implemented all the change suggested. Please, review.